### DISCUSSION TITLE
EventListener do the action many times when using wire:navigate

### DISCUSSION THREAD

**Discussion by Misagh90** [livewire/livewire#6145] [1. Help]
Hi,
I create a like button for my page when I try to like or add to cart & run my "EventListener" it shows a notification but when I click links which have "wire:navigate" and return to that page, the notification again runs the number of clicks I made on wire:navigate.
my code:
```
import toastr from 'toastr'
window.addEventListener('notif', e => {
	toastr.options = {
		"closeButton": true,
		"newestOnTop": false,
		"rtl": true,
		"positionClass": "toast-top-left",
		"maxOpened": 1,
		"preventDuplicates":1,
		"autoDismiss": true,
		"showEasing": "swing",
		"hideEasing": "linear",
		"showMethod": "fadeIn",
		"hideMethod": "fadeOut"
	}
  toastr[e.detail[0].color](e.detail[0].text)
})
```
in my component:
`return $this->dispatch('notif', ['color' => 'success', 'text' => 'item added to cart!']);`

my wire navigate links:
```
		<li>
			<a href="{{ route('storyHolder') }}" wire:navigate class="{{ Request::segment(1) ==  'story' ? 'active' : '' }}">
				<i class="fa-solid fa-book"></i>
				<span>books</span>
			</a>
		</li>
		<li>
			<a href="{{ route('productHolder') }}" wire:navigate class="{{ Request::segment(1) ==  'products' ? 'active' : '' }}">
				<i class="fa-solid fa-shop"></i>
				<span>products</span>
			</a>
		</li>
```
![Screenshot 2023-08-01 153122](https://github.com/livewire/livewire/assets/6697222/1c04d592-f55b-445d-9469-a3e0bddca0ff)

how it'll be fixed?

**Comment by joshhanley** [comment]
@Misagh90 thanks for reporting! Any chance you can provide a component class and view that we can use to investigate this issue? Thanks!
